src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestOptions.cs

```diff
@@ -9,9 +10,13 @@ namespace System.Net.Http
     /// <summary>
     /// Represents a collection of options for an HTTP request.
     /// </summary>
-    public sealed class HttpRequestOptions : IDictionary<string, object?>
+    public sealed class HttpRequestOptions : IDictionary<string, object?>, IReadOnlyDictionary<string, object?>
     {
         private Dictionary<string, object?> Options { get; } = new Dictionary<string, object?>();
+        public bool TryGetValue(string key, out object? value) => Options.TryGetValue(key, out value);
+        public object? this[string key] => Options[key];
+        public IEnumerable<string> Keys => Options.Keys;
+        public IEnumerable<object?> Values => Options.Values;
```

Comment:
These should all be implemented explicitly rather than implicitly:
```suggestion
        bool IReadOnlyDictionary<string, object?>.TryGetValue(string key, out object? value) => Options.TryGetValue(key, out value);
        object? IReadOnlyDictionary<string, object?>.this[string key] => Options[key];
        IEnumerable<string> IReadOnlyDictionary<string, object?>.Keys => Options.Keys;
        IEnumerable<object?> IReadOnlyDictionary<string, object?>.Values => Options.Values;
```